src/entity.ts

```diff
@@ -579,7 +579,16 @@ export namespace entity {
     }
 
     if (typeof value === 'number') {
-      if (value % 1 === 0) {
+      if (Number.isInteger(value)) {
+        if (!Number.isSafeInteger(value)) {
+          process.emitWarning(
+            'Integer value ' +
+              value +
+              ' is outside of bounds of a JavaScript Number\n' +
```

Comment:
Just a thought on some improvements of the messaging:

"Integer value 922337203685477500 is outside of bounds of a JavaScript Number
Use 'Datastore.int(<integer_value_as_string>)' to preserve accuracy during the upload. IntegerOutOfBoundsWarning"

Could be:

"IntegerOutOfBoundsWarning: The value provided is out of bounds of 'Number.MAX_SAFE_INTEGER'.
Use 'Datastore.int(<integer_value_as_string>)' to preserve accuracy during the upload."

We shouldn't include the given integer value in the log, since it won't match what the user actually gave, as the JS engine truncates it:

```js
console.log(9223372036854775807) // 9223372036854776000
```

Is there a way to emit this warning when we also know the property the value comes from, e.g. "The value for `userId` is out of bounds"?

Reply:
Done 👍 

Changed warning message to 
```
IntegerOutOfBoundsWarning: the value for 'largeInt' property is outside of bounds of a JavaScript Number.
Use 'Datastore.int(<integer_value_as_string>)' to preserve accuracy during the upload.
```